epochtalk/epochtalk
# Webpack

Installed webpack and refactored all the frontend code to be webpack compliant. This involves moving all the html files to a public dir and using require.ensure to chunk out whole pages so that they can be pulled down by the client only when they are required.

The build scripts were all updated to use webpack to build for dev and prod environments.

TESTING PROCEDURES:
 * Build using ```foreman start -f Procfile.dev``` and ensure that the bundle and all it's siblings are created in the public/js dir.
 * Ensure the server is started
 * Browse all pages and ensure functionality is there
 * Ensure that the bundle is not minified and that watching file/rebuilding works
 
 * Build using ```foreman start build``` and then use ```foreman start server``` 
 * Ensure that the server is started
 * Browse all pages and ensure functionality is still there
 * Ensure that the bundle is minified
